bundler/lib/dependabot/bundler/native_helpers.rb

```diff
@@ -24,7 +27,17 @@ def build(script_path)
         private
 
         def timeout_command
-          "timeout -s HUP #{timeout_seconds}" if timeout_seconds
+          "timeout -s HUP #{timeout_seconds}" unless timeout_seconds.zero?
+        end
+
+        def adjust(seconds)
+          return 0 unless seconds
+
+          seconds = seconds.to_i
+          return MAX_SECONDS if seconds > MAX_SECONDS
+          return MIN_SECONDS if seconds < MIN_SECONDS
```

Comment:
You might dig [`Comparable#clamp`](https://docs.ruby-lang.org/en/2.5.0/Comparable.html#method-i-clamp)

```plaintext
irb(main):001:0> max = 100
=> 100
irb(main):002:0> min = 1
=> 1
irb(main):003:0> 150.clamp(min, max)
=> 100
irb(main):004:0> 0.clamp(min, max)
=> 1
irb(main):005:0> 37.clamp(min, max)
=> 37
```

It's not the most popular feature. If it feels unintuitive, what you're doing is probably preferable 🙂

Reply:
TIL: [Comparable#clamp](https://ruby-doc.org/core-3.0.2/Comparable.html#method-i-clamp)